sesame/src/main/scala/org/w3/banana/sesame/SesameTransactor.scala

```diff
@@ -0,0 +1,28 @@
+package org.w3.banana.sesame
+
+import org.openrdf.repository.RepositoryConnection
+import org.w3.banana.{Transactor, SparqlUpdate, RDFStore}
+
+import scala.util.{Failure, Try}
+
+trait SesameTransactor extends Transactor[Sesame,RepositoryConnection] {
+
+  def r[T](conn: RepositoryConnection, body: => T): Try[T] = Try{
+    val result = body
+    conn.close()
```

Comment:
If your read body contains an update(s), it/they will be automatically committed to the DB. So we _have_ to do something to prevent that. As Sesame does not have ReadOnly tx's, we have to find a way to mimick them - one way would be to have a [readonly connection](https://github.com/joshsh/sesametools/blob/master/readonly-sail/src/main/java/net/fortytwo/sesametools/readonly/ReadOnlySailConnection.java)

[BigData](http://wiki.blazegraph.com/wiki/index.php/TxGuide) uses [MVCC](https://en.wikipedia.org/wiki/Multiversion_concurrency_control) and hence, along with Jena, have the same R/RW tx semantics, regardless of how the datastore is set up.